sectionproperties/examples/example_nastran_section.py

```diff
@@ -1,21 +1,21 @@
 import sectionproperties.pre.nastran_sections as nsections
-from sectionproperties.analysis.cross_section import CrossSection
+from sectionproperties.analysis.cross_section import Section
 
 # create a HAT1 section
-geometry = nsections.HAT1Section(DIM1=4.0, DIM2=2.0, DIM3=1.5, DIM4=0.1875, DIM5=0.375)
+geometry = nsections.nastran_hat1(DIM1=4.0, DIM2=2.0, DIM3=1.5, DIM4=0.1875, DIM5=0.375)
 geometry.plot_geometry()  # plot the geometry
 
 # create a mesh with a maximum elemental area of 0.005
-mesh = geometry.create_mesh(mesh_sizes=[0.005])
+geometry.create_mesh(mesh_sizes=[0.005])
 
-section = CrossSection(geometry, mesh)  # create a CrossSection object
+section = Section(geometry, time_info=True)  # create a Section object
 section.display_mesh_info()  # display the mesh information
 section.plot_mesh()  # plot the generated mesh`
 
 # perform a geometric, warping and plastic analysis, displaying the time info
-section.calculate_geometric_properties(time_info=True)
-section.calculate_warping_properties(time_info=True)
-section.calculate_plastic_properties(time_info=True)
+section.calculate_geometric_properties()
+section.calculate_warping_properties()
+section.calculate_plastic_properties()
 
 # print the results to the terminal
 section.display_results()
```

Comment:
This file doesn't run for me, when setting `verbose=True` for the plastic properties I get the following:

```
d = -0.583245798319; f_norm = 1.0
d = 1.416754201681; f_norm = -1.0
d = 0.41675420168099997; f_norm = -0.5798319327731093
d = -0.08324579831900003; f_norm = -0.3333333333333334
d = -0.33324579831900003; f_norm = 0.28291316526610644
d = -0.21847307104627278; f_norm = -0.046294881588999165
d = -0.23461298581900003; f_norm = 0.0
---x-axis plastic centroid calculation converged at -2.34613e-01 in 6 iterations.
d = -2.0; f_norm = 1.0
d = 2.0; f_norm = -1.0
d = 0.0; f_norm = 0.0
---y-axis plastic centroid calculation converged at 0.00000e+00 in 2 iterations.
d = -2.0; f_norm = 1.0
d = 2.0; f_norm = -1.0
TopologyException: found non-noded intersection between LINESTRING (0 -0.208246, -2 -0.208246) and LINESTRING (-2 3.26625e+16, 0 -0.583246) at 0 -0.20824579831900003
```

@connorferster any thoughts?

Reply:
@robbievanleeuwen Looking into this now...

Reply:
Well, I am frustrated to report that it "runs on my machine" and I cannot recreate the TopologyException. This coordinate seems suspect `LINESTRING (-2 3.26625e+16, 0 -0.583246)`. I have no idea why it has a point way out in space!

Let me trying updating my shapely to see if that changes anything...

Reply:
I tried running with shapely versions `1.7.0` and `1.8.0` and both ran for me. @ccaprani Would you be able to try running the example? https://github.com/normanrichardson/section-properties/blob/fix-examples/sectionproperties/examples/example_nastran_section.py

Reply:
@robbievanleeuwen Here is what I did: when I started working on sectionproperties this morning, I pulled all changes from the repo to my v2 branch. I opened Jupyter and in a new notebook running within my sectionproperties conda environment and I copy/pasted the example from Norman's Github repo into the notebook. Is it possible that Norman's repo has out-of-date plastic properties code? But, if that were the case, Norman would have problems running the example too, perhaps...
Anyway, just throwing ideas out there.

Reply:
How odd! I will do a bit of digging tomorrow to see if I can visualise what's going on here!

Reply:
Think i found the culprit!

Proposed fix in #131. Error caused by possible floating point chaos resulting in `create_line_sement` generating lines that do not intersect the geometry. When `vector[0]` is extremely small, in my case it was in the order of 1e-17, the line segment was not picked up as being vertical.

Reply:
I had a look and unfortunately, I was unable to replicate the error on my side. I plan to keep working on the PR topic for now. 

It may be prudent to bring my fork up to speed (with any fixes) before submitting the PR, any tips on how to do this in a smooth manner (the extreme version I can think of for now is a local rebase with a forced push, but this will most likely break a few things at this point)?

Reply:
@normanrichardson I don't think that you need to do a rebase. I think you just need to pull changes from the repo to your local branch (resolving any merge conflicts) and then do a new commit/push and the PR will be updated.